src/compiler/checker.ts

```diff
@@ -10694,6 +10695,11 @@ namespace ts {
                 const key = getFlowCacheKey((<PropertyAccessExpression>node).expression);
                 return key && key + "." + (<PropertyAccessExpression>node).name.text;
             }
+            if (node.kind === SyntaxKind.BindingElement) {
+                const key = node.parent.parent.kind === SyntaxKind.BindingElement ? getFlowCacheKey(node.parent.parent) : isApparentTypePosition(node.parent.parent) ? "@<initializer>" : "<initializer>";
+                const text = getBindingElementNameText(node as BindingElement);
+                return key && text && key + "." + text;

```

Comment:
it sounds like from our discussion at lunch that you came up with a better way to make a key.